health_imaging.py

```diff
@@ -637,3 +642,62 @@ class BIRADS(ModelSQL, ModelView):
 
     def get_rec_name(self, name):
         return '%s: %s' % (self.code, self.classification)
+
+
+class RequestPatientImagingTestStart(ModelView):
+    'Request Patient Imaging Test Start'
+    __name__ = 'gnuhealth.patient.imaging.test.request.start'
+
+    existing_requests_count = fields.Function(
+        fields.Integer('Existing Imaging Test Requests'),
+        'on_change_with_existing_requests_count'
+    )
+
+    existing_requests_description = fields.Function(
+        fields.Char('Existing Imaging Test Requests Description',
+                    states={
+                        'invisible': (Eval('existing_requests_count', 0) == 0),
+                    },
+                    depends=['existing_requests_count']
+                    ),
+        'on_change_with_existing_requests_description'
+    )
+
+    @classmethod
+    def view_attributes(cls):
+        return super().view_attributes() + [
+            ('//image[@name="gnuhealth.warning"]', "states", {
+                'invisible': (Eval('existing_requests_count', 0) == 0),
+            })]
+
+    @fields.depends('patient')
+    def on_change_with_existing_requests_count(self, _name=None):
+        ImagingTestRequest = Pool().get('gnuhealth.imaging.test.request')
+
+        if self.patient:
+            return len(ImagingTestRequest.search([
+                ('patient.id', '=', self.patient.id),
+                ('state', 'in', ['draft', 'requested'])
+            ]))
+        else:
+            return 0
+
+    @fields.depends('existing_requests_count', 'patient')
+    def on_change_with_existing_requests_description(self, _name=None):
+        if self.existing_requests_count == 0:
+            return None
+        elif self.existing_requests_count == 1:
+            return 'There is 1 outstanding imaging requests for the patient'
+        else:
+            return 'There are {} outstanding imaging requests for the patient'.format(self.existing_requests_count)
+
+    def get_existing_requests_icon(self, _name):
```

Comment:
remove?

Reply:
Makes sense to remove if we can directly use the icon name. I couldn't see this method being used anywhere.